src/hooks/useContractsOwned.ts

```diff
@@ -1,10 +1,19 @@
 import { useSnapshot } from 'valtio'
-import ContractsStore from 'stores/ContractsStore'
+import ContractsNetworkStore from 'stores/ContractsStore'
+import Network from 'models/Network'
 
-export default function (store: ContractsStore) {
-  const { addressToTokenIds } = useSnapshot(store)
+export default function (networkName?: Network) {
+  const { networks } = useSnapshot(ContractsNetworkStore)
 
-  if (!addressToTokenIds) return []
+  const addressToTokenIds = (Object.keys(networks) as Network[])
+    .filter((network) => !networkName || networkName === network)
```

Comment:
Not sure what is faster, doing `filter` + `reduce`, or doing:
```TS
.reduce((combined, netwrokName) => {
  if (networkName || !(networkName === networkName)) return
  return {
        ...combined,
        ...networks[networkName].addressToTokenIds,
      }
}
```
Something like this, I mean having `if` statement inside `reduce` 🤔

Reply:
here the processing speed is not important